dali/kernels/imgproc/convolution/cutlass/threadblock/predicated_tile_iterator.h

```diff
@@ -182,7 +326,10 @@ class PredicatedTileIterator<Shape_, Element_, layout::PitchLinear, AdvanceRank,
   /// Parameters object is precomputed state and is host-constructible
   class Params {
    public:
-    friend PredicatedTileIterator;
+    friend PositionPredicatedTileIterator;
+    int window_size_;
+    int window_anchor_;
+    int channels_;
```

Comment:
If it doesn't wreak havoc on register usage, use fast_div.

Reply:
Tried with fast_div, and didn't observe any benefit in using that. I removed the switch from the Channels as indeed it added some overhead instead of helping the compiler.